modules/auth/main.tf

```
@@ -16,8 +16,11 @@
 
 locals {
   cluster_ca_certificate = data.google_container_cluster.gke_cluster.master_auth != null ? data.google_container_cluster.gke_cluster.master_auth[0].cluster_ca_certificate : ""
-  endpoint               = data.google_container_cluster.gke_cluster.endpoint != null ? data.google_container_cluster.gke_cluster.endpoint : ""
-  host                   = data.google_container_cluster.gke_cluster.endpoint != null ? "https://${data.google_container_cluster.gke_cluster.endpoint}" : ""
+  private_endpoint       = try(data.google_container_cluster.gke_cluster.private_cluster_config[0].private_endpoint, "")
+  public_endpoint        = try(data.google_container_cluster.gke_cluster.private_cluster_config[0].public_endpoint, "")
+  default_endpoint       = data.google_container_cluster.gke_cluster.endpoint != null ? data.google_container_cluster.gke_cluster.endpoint : ""
+  endpoint               = var.use_private_endpoint == true ? local.private_endpoint : local.public_endpoint != "" ? local.public_endpoint : local.default_endpoint
```

Shouldn't we return the default endpoint if you don't express a preference?

```suggestion
  endpoint               = var.use_private_endpoint == true ? local.private_endpoint : local.public_endpoint != "" ? local.public_endpoint : local.default_endpoint
```

---

I thought about this as well. It might lead to unexpected behaviour though.

E. g. you provide `use_private_endpoint = true`. No `private_endpoint` is found, and as a result the default endpoint is returned. I think I'd rather have an empty `host` in that case and a failing follow-up request. But I'm open to a change.

---

Right, I'm fine with the behavior when `use_private_endpoint = true`. But when `use_private_endpoint = false` we should simply return the default endpoint (not necessarily the public endpoint).

---

Agree, this will also guarantee backwards compatibility.